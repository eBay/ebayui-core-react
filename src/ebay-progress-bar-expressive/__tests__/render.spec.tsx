import React from 'react';
import { composeStories } from '@storybook/react';
import * as stories from './index.stories';
import { render } from '@testing-library/react';

const { Default, WithMessages, WithSingleMessage, WithLongMessage, WithCustomTiming, WithMediumSize } = composeStories(stories);

jest.mock('../../common/random-id')

describe('<EbayProgressBarExpressive /> rendering', () => {
    it('renders default story correctly', () => {
        const { container } = render(<Default />);
        expect(container).toMatchSnapshot();
    });

    it('renders with messages story correctly', () => {
        const { container } = render(<WithMessages />);
        expect(container).toMatchSnapshot();
    });

    it('renders with single message story correctly', () => {
        const { container } = render(<WithSingleMessage />);
        expect(container).toMatchSnapshot();
    });

    it('renders with long message story correctly', () => {
        const { container } = render(<WithLongMessage />);
        expect(container).toMatchSnapshot();
    });

    it('renders with custom timing story correctly', () => {
        const { container } = render(<WithCustomTiming />);
        expect(container).toMatchSnapshot();
    });

    it('renders with medium size story correctly', () => {
        const { container } = render(<WithMediumSize />);
        expect(container).toMatchSnapshot();
    });
});
