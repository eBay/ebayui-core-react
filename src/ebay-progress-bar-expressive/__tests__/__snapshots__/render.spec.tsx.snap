// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EbayProgressBarExpressive /> rendering renders default story correctly 1`] = `
<div>
  <div
    class="progress-bar-expressive"
  >
    <div
      aria-describedby="abc123"
      class="progress-bar-expressive__progress"
      role="progressbar"
    >
      <div
        class="progress-bar-expressive__lines"
      >
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<EbayProgressBarExpressive /> rendering renders with custom timing story correctly 1`] = `
<div>
  <div
    class="progress-bar-expressive"
  >
    <div
      class="progress-bar-expressive__messages"
    >
      <div
        aria-hidden="true"
        class="progress-bar-expressive__message"
        duration="2000"
      >
        Message with 2s duration
      </div>
    </div>
    <div
      aria-describedby="abc123"
      class="progress-bar-expressive__progress"
      role="progressbar"
    >
      <div
        class="progress-bar-expressive__lines"
      >
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<EbayProgressBarExpressive /> rendering renders with long message story correctly 1`] = `
<div>
  <div
    class="progress-bar-expressive"
  >
    <div
      class="progress-bar-expressive__messages"
    >
      <div
        aria-hidden="true"
        class="progress-bar-expressive__message"
      >
        Messages should be one line...
      </div>
    </div>
    <div
      aria-describedby="abc123"
      class="progress-bar-expressive__progress"
      role="progressbar"
    >
      <div
        class="progress-bar-expressive__lines"
      >
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<EbayProgressBarExpressive /> rendering renders with medium size story correctly 1`] = `
<div>
  <div
    class="progress-bar-expressive"
  >
    <div
      class="progress-bar-expressive__messages progress-bar-expressive__messages--medium"
    >
      <div
        aria-hidden="true"
        class="progress-bar-expressive__message"
      >
        Medium Size Message
      </div>
      <div
        class="progress-bar-expressive__message progress-bar-expressive__message--in progress-bar-expressive__message--initial"
        id="abc123"
        role="status"
      >
        Medium Size Message
      </div>
    </div>
    <div
      aria-describedby="abc123"
      class="progress-bar-expressive__progress"
      role="progressbar"
    >
      <div
        class="progress-bar-expressive__lines"
      >
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<EbayProgressBarExpressive /> rendering renders with messages story correctly 1`] = `
<div>
  <div
    class="progress-bar-expressive"
  >
    <div
      class="progress-bar-expressive__messages"
    >
      <div
        aria-hidden="true"
        class="progress-bar-expressive__message"
      >
        Hang tight
      </div>
    </div>
    <div
      aria-describedby="abc123"
      class="progress-bar-expressive__progress"
      role="progressbar"
    >
      <div
        class="progress-bar-expressive__lines"
      >
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<EbayProgressBarExpressive /> rendering renders with single message story correctly 1`] = `
<div>
  <div
    class="progress-bar-expressive"
  >
    <div
      class="progress-bar-expressive__messages"
    >
      <div
        aria-hidden="true"
        class="progress-bar-expressive__message"
      >
        Single Message
      </div>
    </div>
    <div
      aria-describedby="abc123"
      class="progress-bar-expressive__progress"
      role="progressbar"
    >
      <div
        class="progress-bar-expressive__lines"
      >
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
        <div
          class="progress-bar-expressive__line"
        />
      </div>
    </div>
  </div>
</div>
`;
